LeaderboardBackend/Services/Impl/ModshipService.cs

```diff
@@ -30,8 +30,14 @@ public async Task<List<Modship>> LoadUserModships(Guid userId)
 					.ToListAsync();
 	}
 
-	// TODO: Implement this
-	// public async Task DeleteModship(Modship modship)
-	// {
-	// }
+	public async Task DeleteModship(Modship modship)
+	{
+		var deleteModship = _applicationContext.Modships.Where(m =>
+			m.LeaderboardId == modship.LeaderboardId &&
+			m.UserId == modship.UserId)
+			.FirstOrDefault() ?? throw new NullReferenceException();
```

Comment:
I don't know if `NullReferenceException` is the way to go here, cause I don't think it accurately describes the issue. The closest exception I can think of in the `System` namespace is `KeyNotFoundException` but maybe a custom exception would be a good idea for this? Trying to think critically because in all likelihood, all our future Delete operations will be based on whatever you do here.

Reply:
Probably obsoleted by the new requested change